spdx/file.py

```diff
@@ -9,26 +9,47 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from enum import Enum, auto
 from functools import total_ordering
 import hashlib
 
 from spdx import checksum
-from spdx import license
 from spdx import utils
+from spdx.license import License
+from spdx.parsers.builderexceptions import SPDXValueError
 
 
-class FileType(object):
-    SOURCE = 1
-    BINARY = 2
-    ARCHIVE = 3
-    OTHER = 4
-    APPLICATION = 5
-    AUDIO = 6
-    IMAGE = 7
-    TEXT = 8
-    VIDEO = 9
-    DOCUMENTATION = 9
-    SPDX = 10
+class FileType(Enum):
+    SOURCE = auto()
+    BINARY = auto()
+    ARCHIVE = auto()
+    OTHER = auto()
+    APPLICATION = auto()
+    AUDIO = auto()
+    IMAGE = auto()
+    TEXT = auto()
+    DOCUMENTATION = auto()
+    SPDX = auto()
+    VIDEO = auto()
+
+
+def file_type_from_rdf(rdf_file_type: str) -> FileType:
+    """e.g. convert fileType_source to FileType.SOURCE"""
+    file_type_str = rdf_file_type.split("_")[1].upper()
+
+    for file_type_enum in FileType:
+        if file_type_str == file_type_enum.name:
+            file_type = file_type_enum
+            break
```

Comment:
nit: I think we can do this in a nicer way: I learned from Meret recently that you can check whether a string is in the enum names in an elegant way: https://github.com/spdx/tools-python/pull/311/files#diff-ef6ea2e9b61cee128733a6c5ea2c4eb463e9a8a732e532ce8b18f87400e187e6R52-R53. 
Then you can simply get the enum constant like this: https://stackoverflow.com/a/41409392

Reply:
done